app/code/Magento/Quote/Model/Cart/Totals/ItemConverter.php

```diff
@@ -79,7 +79,6 @@ public function modelToDataObject($item)
         $this->eventManager->dispatch('items_additional_data', ['item' => $item]);
         $items = $item->toArray();
         $items['options'] = $this->getFormattedOptionValue($item);
-        unset($items[ExtensibleDataInterface::EXTENSION_ATTRIBUTES_KEY]);
```

Comment:
These are removed on purpose, as it causes a fatal error due to types mismatch on checkout when extension attribute is populated. There is more detailed explanation of what is happening [in this comment to issue 3640](https://github.com/magento/magento2/issues/3640#issuecomment-192291078).
We are currently looking into a solution which may be safely used in this case. I will update the ticket when we come up with one.